src/main/java/org/spongepowered/api/world/extent/Extent.java

```diff
@@ -526,4 +526,16 @@ default void setNotifier(Vector3i pos, @Nullable UUID uuid) {
      */
     Set<AABB> getIntersectingCollisionBoxes(Entity owner, AABB box);
 
+    /**
+     * Creates a new archetype volume from the specified section of this extent.
+     * The archetype's volume will be shifted such that the position given in
+     * the origin will be the origin of the volume.
+     *
+     * @param min The minimum point of the area to copy
+     * @param max The maximum point of the area to copy
+     * @param origin The eventual origin on the new archetype volume
```

Comment:
I think the origin shift should be moved to an ArchetypeVolume method. It's not strictly necessary for creating it, and also someone might want to change it latter. An other option is just to offset it so that the `min` corner is equivalent to `0`.